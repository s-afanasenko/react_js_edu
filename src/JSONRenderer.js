import React from 'react';
import { withTheme } from './theme-context';

class JSONRenderer extends React.Component {
	render () {
		let JSONString;

		switch (typeof this.props.JSONString) {
			case 'string':
				JSONString = this.props.JSONString;
				break;

			default:
				JSONString = JSON.stringify(this.props.JSONString, null, 2);
				break;
		}

		return (
			<pre className={`card bg-${this.props.theme}`}>
				<div className="card-body">
					{ JSONString }
				</div>
			</pre>
		)
	}
}

JSONRenderer.defaultProps = {
	theme: 'light'
}

const WithThemeJSONRenderer = withTheme(JSONRenderer);

export { WithThemeJSONRenderer as JSONRenderer };

export default JSONRenderer;
