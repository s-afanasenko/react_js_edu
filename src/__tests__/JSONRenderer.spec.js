import React from 'react';
import { render } from '@testing-library/react';
import JSONRenderer from '../JSONRenderer';

const objProp = {
	key01: 'value01',
	key02: {
		nestedKey01: 123
	},
	key03: [1, 'string', { key: true }]
}

const stringProp = JSON.stringify(objProp);

describe('<JSONRenderer />', () => {
	it('should match snapshot with string prop', () => {
		const { container } = render(
			<JSONRenderer JSONString={stringProp} />
		);

		expect(container).toMatchSnapshot(); 
	});

	it('should match snapshot with object prop', () => {
		const { container } = render(
			<JSONRenderer JSONString={objProp} />
		);

		expect(container).toMatchSnapshot(); 
	});
});
