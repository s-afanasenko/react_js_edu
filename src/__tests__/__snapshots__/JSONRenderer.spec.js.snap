// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JSONRenderer /> should match snapshot with object prop 1`] = `
<div>
  <pre
    class="card bg-primary"
  >
    <div
      class="card-body"
    >
      {
  "key01": "value01",
  "key02": {
    "nestedKey01": 123
  },
  "key03": [
    1,
    "string",
    {
      "key": true
    }
  ]
}
    </div>
  </pre>
</div>
`;

exports[`<JSONRenderer /> should match snapshot with string prop 1`] = `
<div>
  <pre
    class="card bg-primary"
  >
    <div
      class="card-body"
    >
      {"key01":"value01","key02":{"nestedKey01":123},"key03":[1,"string",{"key":true}]}
    </div>
  </pre>
</div>
`;
